packages/dataviews/src/dataform.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import type { FormEventHandler } from 'react'; // TODO: how to import FormEventHandler?
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	TextControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useCallback, useMemo, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { NormalizedField } from './types';
+
+type DataFormProps< Item > = {
+	item: Item;
+	onUpdateItem: ( item: Item ) => void;
+	fields: NormalizedField< Item >[]; // TODO: use Field. Normalize them first.
+	form: {
```

Comment:
Would `metadata` be a better name for this? 

`form` seems to imply this is contains the actual form fields.

Reply:
As per https://github.com/WordPress/gutenberg/issues/59745, I think we want to call it form, like we did in DataViews (e.g.: `<DataViews view={view} />`, `<DataForm form={form} />`).

Reply:
Actually for me the `form` object as I imagined it is less about these event handlers `onChange`, `onSubmit`...  and more about the "layout" of the form, which fields are visible, how they are groups...

Seeing, what's currently in the form object, It raised the question about whether all the submit button and form submission handling should actually be part of `DataForm` or not.

I'm leaning towards no for the moment, for instance in the "details panel" case, there's no submit button. So I see `DataForms` as something you actually can use within a form but not the whole form. I see it as a group of `*Control` components basically. If we simplify, I see it as a component that has a value (data here) and a single `onChange` handler, all the rest is left to the wrapper component (submit...)

Reply:
I iterated on this and moved back the submit/cancel actions to the consumer.

I agree `form` is about the layout. If we wanted to absorb the form actions (submit/cancel) plus the corresponding labels, a better API could be:

```js
<DataForm
  data={ data }
  fields={fields}
  form={ form }
  onUpdate={ onUpdate }
  /* Absorb submit/cancel */
  labels={ labels }
  onSubmit={ onSubmit }
  onCancel={ onCancel }
  />
```

For an use case like the details panel, the interaction model suggested `onUpdate` would be triggered for every change. Though for modals like in this PR's use case, I went back and forth about triggering the update for every change vs only triggering upon submitting. The former seems a simpler approach, I didn't like we had to absorb the `event` handler for forms.